Add project configuration

ant31 opened the pull request:
1. It's easier to have a file to manage server-side configuration (configmap) instead set 20+ env vars

ant31 commented:
/retest

ant31 commented:
/retest

ant31 commented:
foo

ant31 commented:
@samba 

The PR enable the configuration via a config file instead of env variable.

It includes some code to prepare follow-up pr for: 
- fetch per-repository configuration
- trigger or re-trigger builds from github via labels / comments
- rules to who can trigger builds (e.g contributors only)

samba commented on deploy/failfast/resources/ff-api.yaml:
```diff
@@ -0,0 +1,75 @@
+            secretKeyRef:
+              key: integration_pem
+              name: failfast-ci
+        image: quay.io/failfast-ci/failfast-cibuild:pr-36-failfast-configuration-8b7ba189
```
The image label here seems like something we'd want to generalize for use in other deployments. Stable? (TODO?)

ant31 commented:
@samba I've removed the default integration_id. 
About the deployment, it's using `kubectl kinflate` and added small documentation on how to use it:
https://github.com/failfast-ci/failfast-api/blob/36e2ee5ad9b3f93d0f2854d0a483da91cf9169a5/deploy/failfast/Readme.md
